kafka-admin/src/main/resources/graphql-schema/kafka-admin.graphql

```diff
@@ -1,11 +1,67 @@
-extend type Query {
-    topic(topicName: String): Topic
-    topicList(filter: String): [Topic]
+type TopicDescription {
+    #id: ID
+    name: String
+
+    """@transient"""
+    partitions: [Partitions]
+    config: TopicConfig
+}
+
+type Partitions {
+    partition: Int
+    replicas: [Replicas]
+}
+
+type Replicas {
+    id: String
+    inSync: Boolean
 }
 
-type Topic {
+type TopicOnlyName {
+    #id: ID
     name: String
+}
+
+type TopicConfig {
+    partitionCount: Int
+    replicationFactor: Int
     isInternal: Boolean
+    # config
+    minInsyncReplicas: Int
+    pairs: [ConfigEntry]
+}
+
+type ConfigEntry {
+    key: String
+    value: String
+}
+
+input CreateOrMutateTopicConfigInput {
+    retentionDays: Int
+    minInSyncReplicas: Int
     partitionCount: Int
     replicationFactor: Int
+    isInternal: Boolean
+}
+
+input CreateTopicInput {
+    name: String
+    config: CreateOrMutateTopicConfigInput
+}
+
+input MutateTopicInput {
+    #id: ID!
+    name: String!
+    config: CreateOrMutateTopicConfigInput
+}
+
+extend type Mutation {
+    createTopic(input: CreateTopicInput!): TopicDescription
+    updateTopic(input: MutateTopicInput!): TopicDescription
+    deleteTopic(name: String!): String
```

Comment:
Rule of thumb is to never ever return Scalar from the API. It will make our API breaking with each change - no ability to build non breaking API

Reply:
Oh. Good to know! Thanks.